identity/src/main/java/com/stripe/android/identity/ui/DocumentScanScreen.kt

```diff
@@ -184,51 +185,68 @@ private fun DocumentCaptureScreen(
     // Throttle the message change speed in Found state
     var foundResId by remember { mutableIntStateOf(R.string.stripe_hold_still) }
     var lastFoundResIdUpdateTime by remember { mutableLongStateOf(0L) }
+    var timeSinceInitalState by remember { mutableLongStateOf(0L) }
+    var shouldShowInitial by remember { mutableStateOf(false) }
```

Comment:
These should probably all be `rememberSaveable`, no? Or better, can they be stored in a state object in the ViewModel?

Reply:
we forced to portrait but replaced with `rememberSaveable`

Currently the `documentScannerState` comes from `identityScanViewModel`, which dictates the scanning state of both document and selfie, but selfie scan does have its own UX logic on rendering different states(there's no throttling there)  

Since this logic is document specific, would leave them here atm rather than refactor `IdentityScanViewModel` for now

Reply:
Can’t we still model this with a `StateFlow` in the ViewModel? I imagine that we can do something like:

```kotlin
val message: StateFlow<String> = documentScannerState.map { /* … */ }
```

Having all this logic in the composable is hard to test, and it’d become much easier once we pull it out into the ViewModel.